src/test/java/tfb/status/handler/ShareResultsViewHandlerTest.java

```diff
@@ -0,0 +1,95 @@
+package tfb.status.handler;
+
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static io.undertow.util.StatusCodes.NOT_FOUND;
+import static io.undertow.util.StatusCodes.OK;
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.io.MoreFiles;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpResponse;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import tfb.status.service.FileStore;
+import tfb.status.service.ShareResultsUploader;
+import tfb.status.testlib.HttpTester;
+import tfb.status.testlib.ResultsTester;
+import tfb.status.testlib.TestServicesInjector;
+import tfb.status.view.Results;
+
+/**
+ * Tests for {@link ShareResultsViewHandler}.
+ */
+@ExtendWith(TestServicesInjector.class)
+public final class ShareResultsViewHandlerTest {
+  /**
+   * Ensure that after uploading a file, we can use the view handler to retrieve
+   * that results file using the GET {@code /share-results/view/{shareId}.json}
+   * endpoint.
+   */
+  @Test
+  public void shareResultsViewHandler_getValidUpload(
+      HttpTester http,
+      ResultsTester resultsTester,
+      ShareResultsUploader shareResultsUploader,
+      FileSystem fileSystem) throws IOException, InterruptedException {
+
+    Results results = resultsTester.newResults();
+
+    Path jsonFile = fileSystem.getPath("results_to_upload.json");
```

Comment:
I have an eventual goal to allow all the tests to run concurrently.  I think this code here may be modeled after an existing test that did something similar using the same filename, but as a result, the two tests would conflict with each other if ran concurrently, wouldn't they?  Could you prepend the test class name and/or test method name to this filename, or figure out some other scheme that's likely to keep this non-conflicting with other tests?

Reply:
I made them use random UUIDs, deos that work? If that's messy or not right I can update to use class name + test method name.

Reply:
That's fine.